### DISCUSSION TITLE
Piping to pager (eg. less -R) loses all color

### DISCUSSION THREAD

**Discussion by rockorager** [charmbracelet/lipgloss#122] [Q&A] [answered]
I must be missing something obvious, but when I pipe any output of lipgloss to a pager, I lose all color formatting. I'm aware I need to use a -R flag on less in order to preserve escape codes. However, I can pipe straight to vim and see that no codes are preserved.

A minimal example:
```go
package main

import (
	"fmt"

	"github.com/charmbracelet/lipgloss"
)

func main() {
	style := lipgloss.NewStyle().Foreground(lipgloss.Color("4"))
	fmt.Println(style.Render("Hello, world"))
}
```

```sh
$ go run ./
```

## Output to terminal
(In blue) Hello, world

```sh
$ go run ./ | less -R
```
## Output to less -R
(No color) Hello, world

## Expected output to less -R
(In blue) Hello, world

Am I missing something??

**Comment by muesli** [comment, marked as answer]
`lipgloss` (or rather `termenv` in the background) detects that the output isn't connected to a terminal and strips the styling. You can either set a color profile manually or set an environment variable which enforces colored output:

```go
lipgloss.SetColorProfile(termenv.TrueColor)
```

```bash
CLICOLOR_FORCE=1 go run .
```

**Reply by rockorager** [reply]
Thanks!

**Comment by beltagyy** [comment]
This is not a bug -- it is TTY detection. Lipgloss checks whether stdout is attached to a terminal. When you pipe to a pager, stdout becomes a pipe, so lipgloss strips ANSI escape sequences by default.\n\nTo force color through a pipe:\n\n1. Environment variable (quickest):\nCLICOLORS_FORCE=1 go run . | less -R\n\n2. Explicit color profile in code (v2):\nimport github.com/charmbracelet/colorprofile\nlipgloss.SetColorProfile(colorprofile.TrueColor)\nstyle := lipgloss.NewStyle().Foreground(lipgloss.Color("4"))\nfmt.Println(style.Render("Hello, world"))\n\n3. Per-renderer basis:\nrenderer := lipgloss.NewRenderer(os.Stdout, colorprofile.TrueColor)\nfmt.Println(renderer.Render(style.Render("Hello, world")))\n\nMethod 1 works for ad-hoc pager use. Method 2/3 gives deterministic output for CLI tools that pipe by design.
